models/accm.py

```diff
@@ -38,12 +38,13 @@ def __init__(self, input_layer, h_channels, number_of_rsacm):
             [(RsacmBlock(input_layer, self.h_channels)) for i in range(0, self.number_of_rsacm)])
 
     def forward(self, x):
+        x = x.detach() # Detach x make sure that gradient for this layer won't flow to previous layers.
         self.x_list = []
         x_copy = x.clone()

```

Comment:
I think that these clone() calls are unnecessary, unless in place operations are used on exactly these tensors (I think there are none). clone() makes an actual copy in memory, and it is more efficient to just hold a reference to the original tensor to keep it alive.